shared/src/sources/balancer/swap/stable_math.rs

```diff
@@ -0,0 +1,485 @@
+// TODO - incorporate this into baseline solver.
+#![allow(dead_code)]
+use super::error::Error;
+use crate::sources::balancer::swap::fixed_point::Bfp;
+use crate::sources::balancer::swap::math::{rounded_div, BalU256};
+use ethcontract::U256;
+use lazy_static::lazy_static;
+
+lazy_static! {
+    static ref AMP_PRECISION: U256 = U256::from(1000);
+}
+
+fn calculate_invariant(
+    amplification_parameter: U256,
+    balances: Vec<Bfp>,
```

Comment:
Can be `&[Bfp]` so that you don't have to clone this when calling the function.

Reply:
Great idea - Thanks!